src/stdune/csexp.mli

```diff
@@ -0,0 +1,15 @@
+type t = Sexp.t
+
+exception Parse_error of string
```

Comment:
If we plan to handle such errors, we should make `parse_canonical` return a `result`. Otherwise, we should just use `User_error.raise`